### PR TITLE
Add `(host|control-plane).minikube.internal` to /etc/hosts

### LINKED ISSUE #7246: ControlPlane IP changes make cluster restarts less reliable
While doing testing of Kubernetes upgrades since v1.18, I noticed that:

* Clusters no longer come up naturally if the VM IP changes
* Upgrades no longer operate as smoothly if the VM IP changes

Eventually the cluster does come up, but only once all of the control plane services go into a crash loop.

### PR DESCRIPTION
**Pull request by tstromberg**
Adds two new hostname aliases that are available to all nodes:

* control-plane.minikube.internal - The ControlPlane frontend
* host.minikube.internal - The VM Host IP 

The first is used so that nodes have consistent connectivity to the apiserver across IP changes. Fixes #7246 -- this means that our configurations do not have to be rewritten when the IP changes.

The second is used so that users have an easier time accessing host services. This will allow us to improve this workflow: https://minikube.sigs.k8s.io/docs/handbook/host-access/

### CODE REVIEW

**Comment by medyagh** [comment]
```
Docker_Linux — Jenkins: completed with 10 / 30 failures in 3.40 minute(s).
```
seems much higher than HEAD (which is at 6)

**Comment by sharifelgamal** [comment]
The docker tests are panicking on the dashboard test:
```
22:12:10 2020/03/25 22:12:10 [ERR] GET %2A%20This%20control%20plane%20is%20not%20running%21%20%28state=Stopped%29 request failed: Get %2A%20This%20control%20plane%20is%20not%20running%21%20%28state=Stopped%29: unsupported protocol scheme ""
22:12:10 panic: runtime error: invalid memory address or nil pointer dereference [recovered]
22:12:10 	panic: runtime error: invalid memory address or nil pointer dereference
22:12:10 [signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x14b43b6]
22:12:10 
22:12:10 goroutine 238 [running]:
22:12:10 testing.tRunner.func1(0xc0001f1200)
22:12:10 	/usr/local/go/src/testing/testing.go:874 +0x3a3
22:12:10 panic(0x1643fe0, 0x267ac90)
22:12:10 	/usr/local/go/src/runtime/panic.go:679 +0x1b2
22:12:10 k8s.io/minikube/test/integration.validateDashboardCmd(0x1accb00, 0xc00059e060, 0xc0001f1200, 0xc0002fa0c0, 0x2a)
22:12:10 	/var/lib/jenkins/go/src/k8s.io/minikube/test/integration/functional_test.go:339 +0x2f6
22:12:10 k8s.io/minikube/test/integration.TestFunctional.func3.1(0xc0001f1200)
22:12:10 	/var/lib/jenkins/go/src/k8s.io/minikube/test/integration/functional_test.go:130 +0x97
22:12:10 testing.tRunner(0xc0001f1200, 0xc0002486c0)
22:12:10 	/usr/local/go/src/testing/testing.go:909 +0xc9
22:12:10 created by testing.(*T).Run
22:12:10 	/usr/local/go/src/testing/testing.go:960 +0x350
```

**Comment by sharifelgamal** [comment]
The test data yaml files need to be changed for the unit tests to pass, these files are easily the most cumbersome things to maintain in this codebase.

File: pkg/minikube/machine/start.go [hunk 1]
```diff
@@ -288,3 +289,18 @@ func showHostInfo(cfg config.ClusterConfig) {
...
+
+// AddHostAlias makes fine adjustments to pod resources that aren't possible via kubeadm config.
+func AddHostAlias(c command.Runner, name string, ip net.IP) error {
+	glog.Infof("checking")
+	record := fmt.Sprintf("%s\t%s", ip, name)
+	if _, err := c.RunCmd(exec.Command("grep", record+"$", "/etc/hosts")); err == nil {
+		return nil
+	}
+
+	script := fmt.Sprintf(`{ grep -v '\t%s$' /etc/hosts; echo "%s"; } > /tmp/h.$$; sudo mv /tmp/h.$$ /etc/hosts`, name, record)
```
Comment (sharifelgamal) [inline pkg/minikube/machine/start.go]: this bit of bash is a little dense, can you add a comment about what it's doing? 

it looks like it's replacing an existing record in /etc/hosts with a new minikube specific one, but I'm not 100% sure

Comment (medyagh) [reply pkg/minikube/machine/start.go]: agree with @sharifelgamal

**Comment by tstromberg** [comment]
This is now ready for code-review again!

**Comment by tstromberg** [comment]
Test failures appear to be due to #7874 

```
Warning  FailedCreatePodSandBox  4m47s                kubelet, containerd-20200424031633-5224  Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "5f9542ae0caf3e5f191eaa74958a13551287d4181f828645772f564dbc74b8f7": unsupported CNI result version "0.4.0"
```

**Comment by medyagh** [comment]
KVM test looks scary but maybe they will go away if you merge with upstream

**Pull request merged**